src/cursor/cur_layered.c

```diff
@@ -666,6 +673,78 @@ __clayered_reset(WT_CURSOR *cursor)
     API_END_RET(session, ret);
 }
 
+/*
+ * __clayered_copy_constituent_bound --
+ *     Copy the top level bound into a single constituent cursor
+ */
+static int
+__clayered_copy_constituent_bound(WT_CURSOR_LAYERED *clayered, WT_CURSOR *constituent)
+{
+    WT_CURSOR *base_cursor;
+    WT_SESSION_IMPL *session;
+
+    session = CUR2S(clayered);
+    base_cursor = (WT_CURSOR *)clayered;
+
+    if (constituent == NULL)
+        return (0);
+
+    /* Copy across all the bound configurations */
+    F_SET(constituent, F_MASK(base_cursor, WT_CURSTD_BOUND_ALL));
+    /* Note that the inclusive flag is additive to upper/lower, so no need to check it as well */
+    if (F_ISSET(base_cursor, WT_CURSTD_BOUND_UPPER))
+        WT_RET(__wt_buf_dup(session, &constituent->upper_bound, &base_cursor->upper_bound));
```

Comment:
Should we clear the flag on failures? I think we should copy the flags after we successfully set the bound.

Reply:
Done - good idea!